prog/encodingexec.go

```diff
@@ -54,8 +64,9 @@ func (p *Prog) SerializeForExec(buffer []byte, pid int) error {
 		foreachArg(c, func(arg, _ *Arg, _ *[]*Arg) {
 			if arg.Kind == ArgPointer && arg.Res != nil {
 				foreachSubargOffset(arg.Res, func(arg1 *Arg, offset uintptr) {
-					if len(arg1.Uses) != 0 {
-						w.args[arg1] = argInfo{Offset: offset}
+					w.args[arg1] = argInfo{Addr: physicalAddr(arg) + offset}
+					if arg1.Kind == ArgGroup || arg1.Kind == ArgUnion {
```

Comment:
I've added it since I changed foreachSubargOffset to iterate over structs and unions. It is probably mutually exclusive, however it's not obvious. I think it makes sense to leave this check for clarity.